### ISSUE TITLE
There may be no complete callback when download the picture of the local path

### ISSUE THREAD

**Issue by xiaoxiaoyingqian** [SDWebImage/SDWebImage#2910]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           |  ios 
 Platform Version        | 13.0
 SDWebImage Version      | 4.4.7
 Integration Method      | cocoapods
 Xcode Version           | Xcode 11
 Repro rate              | sometimes <10%
 Repro with our demo prj |  
 Demo project link       | 

### Issue Description and Steps
load image by - (void)sd_setImageWithURL:(nullable NSURL *)url,
If the URL is a local path, for example: file: / / / var /..., sometimes, there is no callback ,and imagview set image failure 
fix: 
I found a bit of code in the SDWebimageDownloader.m file,
This method
```objectivec
- (nullable SDWebImageDownloadToken *)downloadImageWithURL:(nullable NSURL *)url
options:(SDWebImageDownloaderOptions)options
progress:(nullable SDWebImageDownloaderProgressBlock)progressBlock
completed:(nullable SDWebImageDownloaderCompletedBlock)completedBlock {
```
when !operation || operation.isFinished || operation.isCancelled will call
```objectivec
[self.downloadQueue addOperation:operation]
```
then call
```objectivec
[operation addhandlersforprogress: progressblock completed: completedblock];
```
if the download is fast enough, or local file, URLSession didCompleteWithError will call before adding block of callback,there will be no callback
I fix it by add block of callback befor addOperation to downloadQueue
```objectivec
if (!operation || operation.isFinished || operation.isCancelled) {
        operation = [self createDownloaderOperationWithUrl:url options:options];
        __weak typeof(self) wself = self;
        operation.completionBlock = ^{
            __strong typeof(wself) sself = wself;
            if (!sself) {
                return;
            }
            LOCK(sself.operationsLock);
            [sself.URLOperations removeObjectForKey:url];
            UNLOCK(sself.operationsLock);
        };
        [self.URLOperations setObject:operation forKey:url];
        // Add operation to operation queue only after all configuration done according to Apple's doc.
        // `addOperation:` does not synchronously execute the `operation.completionBlock` so this will not cause deadlock.
        downloadOperationCancelToken = [operation addHandlersForProgress:progressBlock completed:completedBlock];
        [self.downloadQueue addOperation:operation];
    }
```

**Comment by dreampiggy** [comment]
@xiaoxiaoyingqian You means, you change the call order of that two methods calls, and this issue been solved ?

```objective-c
downloadOperationCancelToken = [operation addHandlersForProgress:progressBlock completed:completedBlock];
[self.downloadQueue addOperation:operation];
```

->

```objective-c
downloadOperationCancelToken = [operation addHandlersForProgress:progressBlock completed:completedBlock];
[self.downloadQueue addOperation:operation];
```

Could you create a PR ? Or I fix this and release new patch version for this ? Seems a issue that may effect some of users.

**Comment by dreampiggy** [comment]
But actually, the `NSOperationQueue addOperation:` is **Always execute the operation async, not sync**. I don't find any effect between these two changes...But if this can solve the problem, maybe we can fix this.

**Comment by xiaoxiaoyingqian** [comment]
yes, the method addOperation is asynchronously , but if call addHandlersForProgress... first, we can ensure that the session's callback must be later than addHandlersForProgress...,just like
```objectc
//[self.downloadQueue addOperation:operation];
   dispatch_async(dispatch_get_global_queue(0, 0), ^{
        NSLog(@"========sessionCallBack");
    });
    NSInteger i = 0;
    while (i++ < 1000000) {
        if (i == 1000000) {
//downloadOperationCancelToken = [operation addHandlersForProgress:progressBlock completed:completedBlock];
            NSLog(@"========addHandlersForProgress");
        }
    }
```
sometimes, "========sessionCallBack" printed earlier than "========addHandlersForProgress"

**Comment by dreampiggy** [comment]
@xiaoxiaoyingqian Maybe we need a PR. Fix for both 4.x branch (v4.4.8 version) and 5.x branch (v5.4.1).

Do you want to create one by yourself, or I create one for this ?
